src/blk/BlockDevice.h

```diff
@@ -175,7 +175,8 @@ class BlockDevice {
   static block_device_t device_type_from_name(const std::string& blk_dev_name);
   static BlockDevice *create_with_type(block_device_t device_type,
     CephContext* cct, const std::string& path, aio_callback_t cb,
-    void *cbpriv, aio_callback_t d_cb, void *d_cbpriv);
+    void *cbpriv, aio_callback_t d_cb, void *d_cbpriv, const char* type);
```

Comment:
IMO better rename this parmeter (and all other similar ones below) to dev_name as we have another *_type in the list